### ISSUE TITLE
pgo-scheduler keeps triggering jobs after pgo delete schedule

### ISSUE THREAD

**Issue by aurelienmarie** [CrunchyData/postgres-operator#1215]
**Describe the bug**
Once you have deleted a schedule using `pgo delete schedule mycluster`, we can see that the configmap has been in fact deleted. However, the job keeps being triggered by the pgo-scheduler.

**To Reproduce**
1. `pgo create cluster mycluster`
2. `pgo create schedule --schedule="*/2 * * * *" --schedule-type=pgbackrest --pgbackrest-backup-type=incr mycluster -n pgouser1`
3. `pgo show schedule -n pgouser1 mycluster` at this point, we have a corresponding configMap
4. `pgo delete schedule -n pgouser1 mycluster` 
5.  `kubectl get jobs -n pgouser1` : you should see the job getting created every 2 minutes according to cron expression
6. delete operator pod and see no jobs are being triggered again

**Expected behavior**
No more job triggered once the `pgo delete schedule` has been deleted. 
FYI if the operator pod is killed, no jobs will be triggered.

**Screenshots**
```{ pgo }  » k describe pod -n pgo postgres-operator-7db46f467d-8clnx            
Name:         postgres-operator-7db46f467d-8clnx
Namespace:    pgo
Priority:     0
Node:         minikube/192.168.99.100
Start Time:   Mon, 13 Jan 2020 14:44:50 +0100
Labels:       name=postgres-operator
              pod-template-hash=7db46f467d
              vendor=crunchydata
Annotations:  <none>
Status:       Running
IP:           172.17.0.2
IPs:
  IP:           172.17.0.2
Controlled By:  ReplicaSet/postgres-operator-7db46f467d
Containers:
  apiserver:
    Container ID:   docker://fb9770f8a27170840ec2165df0c17557eb0686f14c617a8ac046d97162d7747d
    Image:          crunchydata/pgo-apiserver:centos7-4.2.0
    Image ID:       docker-pullable://crunchydata/pgo-apiserver@sha256:72d98099e5f0521bb84360032c30b5e709440f95939d4be3bc629a57c713052a
    Port:           8080/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Wed, 15 Jan 2020 16:07:52 +0100
    Last State:     Terminated
      Reason:       Error
      Exit Code:    255
      Started:      Wed, 15 Jan 2020 15:52:03 +0100
      Finished:     Wed, 15 Jan 2020 16:07:25 +0100
    Ready:          True
    Restart Count:  2
    Liveness:       http-get http://:8080/healthz delay=15s timeout=1s period=5s #success=1 #failure=3
    Readiness:      http-get http://:8080/healthz delay=15s timeout=1s period=5s #success=1 #failure=3
    Environment:
      CRUNCHY_DEBUG:           true
      PORT:                    8080
      PGO_INSTALLATION_NAME:   devtest
      PGO_OPERATOR_NAMESPACE:  pgo (v1:metadata.namespace)
      TLS_CA_TRUST:
      TLS_NO_VERIFY:           false
      DISABLE_TLS:             true
      NOAUTH_ROUTES:
      ADD_OS_TRUSTSTORE:       false
      DISABLE_EVENTING:        false
      EVENT_ADDR:              localhost:4150
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from postgres-operator-token-vznkm (ro)
  operator:
    Container ID:   docker://44fdfc76f7a96e1a8a59c223bde715f78c08f44535bc0860673044fad1fae850
    Image:          crunchydata/postgres-operator:centos7-4.2.0
    Image ID:       docker-pullable://crunchydata/postgres-operator@sha256:80ca0c854177eeb3421582598f07954bd59c9e6028d49704fd74dfc761527abb
    Port:           <none>
    Host Port:      <none>
    State:          Running
      Started:      Wed, 15 Jan 2020 16:07:53 +0100
    Last State:     Terminated
      Reason:       Error
      Exit Code:    255
      Started:      Wed, 15 Jan 2020 15:52:04 +0100
      Finished:     Wed, 15 Jan 2020 16:07:25 +0100
    Ready:          True
    Restart Count:  2
    Readiness:      exec [ls /tmp] delay=4s timeout=1s period=5s #success=1 #failure=3
    Environment:
      CRUNCHY_DEBUG:           true
      NAMESPACE:               pgouser1,pgouser2
      PGO_INSTALLATION_NAME:   devtest
      PGO_OPERATOR_NAMESPACE:  pgo (v1:metadata.namespace)
      MY_POD_NAME:             postgres-operator-7db46f467d-8clnx (v1:metadata.name)
      DISABLE_EVENTING:        false
      EVENT_ADDR:              localhost:4150
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from postgres-operator-token-vznkm (ro)
  scheduler:
    Container ID:   docker://880da8e368826954b4837f5068f1b83b0d9386417a82631ff3a49ace3b32ee2d
    Image:          crunchydata/pgo-scheduler:centos7-4.2.0
    Image ID:       docker-pullable://crunchydata/pgo-scheduler@sha256:d4a73016063bb155ceacb87650f99841fbf327d04e1778e545c66d11ddf35f19
    Port:           <none>
    Host Port:      <none>
    State:          Running
      Started:      Wed, 15 Jan 2020 16:07:54 +0100
    Last State:     Terminated
      Reason:       Error
      Exit Code:    255
      Started:      Wed, 15 Jan 2020 15:52:20 +0100
      Finished:     Wed, 15 Jan 2020 16:07:25 +0100
    Ready:          True
    Restart Count:  3
    Liveness:       exec [bash -c test -n "$(find /tmp/scheduler.hb -newermt '61 sec ago')"] delay=60s timeout=1s period=60s #success=1 #failure=2
    Environment:
      CRUNCHY_DEBUG:           true
      PGO_OPERATOR_NAMESPACE:  pgo (v1:metadata.namespace)
      PGO_INSTALLATION_NAME:   devtest
      TIMEOUT:                 3600
      EVENT_ADDR:              localhost:4150
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from postgres-operator-token-vznkm (ro)
  event:
    Container ID:   docker://e0cd215a8f03f1d73ad017497a04d120bb44e8ac41c6c02c3ed745beb5eac6a5
    Image:          crunchydata/pgo-event:centos7-4.2.0
    Image ID:       docker-pullable://crunchydata/pgo-event@sha256:a2885d84336b834c1fd56de7c5862c4743691b773763a74691d5929debbc4dc0
    Port:           <none>
    Host Port:      <none>
    State:          Running
      Started:      Wed, 15 Jan 2020 16:07:54 +0100
    Last State:     Terminated
      Reason:       Error
      Exit Code:    255
      Started:      Wed, 15 Jan 2020 15:52:05 +0100
      Finished:     Wed, 15 Jan 2020 16:07:25 +0100
    Ready:          True
    Restart Count:  2
    Liveness:       http-get http://:4151/ping delay=15s timeout=1s period=5s #success=1 #failure=3
    Environment:
      TIMEOUT:  3600
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from postgres-operator-token-vznkm (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             True
  ContainersReady   True
  PodScheduled      True
Volumes:
  postgres-operator-token-vznkm:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  postgres-operator-token-vznkm
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason          Age                From               Message
  ----     ------          ----               ----               -------
  Normal   Pulled          42m                kubelet, minikube  Container image "crunchydata/pgo-apiserver:centos7-4.2.0" already present on machine
  Normal   SandboxChanged  42m                kubelet, minikube  Pod sandbox changed, it will be killed and re-created.
  Normal   Created         42m                kubelet, minikube  Created container apiserver
  Normal   Started         42m                kubelet, minikube  Started container apiserver
  Normal   Pulled          42m                kubelet, minikube  Container image "crunchydata/postgres-operator:centos7-4.2.0" already present on machine
  Normal   Created         42m                kubelet, minikube  Created container operator
  Normal   Started         42m                kubelet, minikube  Started container operator
  Normal   Pulled          42m                kubelet, minikube  Container image "crunchydata/pgo-event:centos7-4.2.0" already present on machine
  Normal   Started         42m                kubelet, minikube  Started container event
  Normal   Created         42m                kubelet, minikube  Created container event
  Warning  BackOff         42m (x2 over 42m)  kubelet, minikube  Back-off restarting failed container
  Normal   Created         42m (x2 over 42m)  kubelet, minikube  Created container scheduler
  Normal   Pulled          42m (x2 over 42m)  kubelet, minikube  Container image "crunchydata/pgo-scheduler:centos7-4.2.0" already present on machine
  Normal   Started         42m (x2 over 42m)  kubelet, minikube  Started container scheduler
  Normal   SandboxChanged  27m                kubelet, minikube  Pod sandbox changed, it will be killed and re-created.
  Normal   Pulled          27m                kubelet, minikube  Container image "crunchydata/pgo-apiserver:centos7-4.2.0" already present on machine
  Normal   Created         27m                kubelet, minikube  Created container apiserver
  Normal   Started         27m                kubelet, minikube  Started container apiserver
  Normal   Pulled          27m                kubelet, minikube  Container image "crunchydata/postgres-operator:centos7-4.2.0" already present on machine
  Normal   Created         27m                kubelet, minikube  Created container operator
  Normal   Started         27m                kubelet, minikube  Started container operator
  Normal   Pulled          27m                kubelet, minikube  Container image "crunchydata/pgo-scheduler:centos7-4.2.0" already present on machine
  Normal   Created         27m                kubelet, minikube  Created container scheduler
  Normal   Started         27m                kubelet, minikube  Started container scheduler
  Normal   Pulled          27m                kubelet, minikube  Container image "crunchydata/pgo-event:centos7-4.2.0" already present on machine
  Normal   Created         27m                kubelet, minikube  Created container event
  Normal   Started         27m                kubelet, minikube  Started container event
```

**Please tell us about your environment:**

* Where is this running : Local
* Storage being used (NFS, Hostpath, Gluster, etc): Hosthpath
* Container Image Tag: centos7-4.2.0
* Platform (Docker, Kubernetes, OpenShift): Minikube
* Platform Version: minikube v1.5.2

**Comment by jkatz** [comment]
Just under the radar before our v4.2.1 release, but we believe we have a fix for this.

Before the proposed patch, some people were able to reproduce the issue, some people were not. Hopefully the race condition fixes in cron.v3 fix the above.

**Comment by jkatz** [comment]
Thanks for the report. This is merged in and will be available in `v4.2.1` which incredibly is being released this week. Please don't expect this kind of turnaround time normally, though we do our best to fix things quickly 😉

**Comment by aurelienmarie** [comment]
Many thanks for your quick reply ! I will be looking forward this new version
